Region/Histogram.cc

```diff
@@ -14,12 +14,13 @@ Histogram::Histogram(int num_bins, float min_value, float max_value, const std::
 Histogram::Histogram(Histogram& h, tbb::split) : _bin_width(h._bin_width), _min_val(h._min_val), _hist(h._hist.size(), 0), _data(h._data) {}
 
 void Histogram::operator()(const tbb::blocked_range<size_t>& r) {
-    std::vector<int> tmp(_hist);
+    std::vector<size_t> tmp(_hist);
```

Comment:
aren't we still using 64-bit bins here, though? The question is whether we definitely need them or not.

Reply:
The type size_t may reduce the performance. May I detect the size of incoming data to switch the type?

Reply:
I think trying to detect the size might make it slower (it will introduce a jump for each assigment. Given that we _already_ send the bins as `int32` in the protocol buffers, lets just stick to `int` here. If we find we need to adjust to 64-bit, we will need to do this for the protocol buffers as well (or switch to variable-length integers)

Reply:
Got it. Fixed them.